src/utils/device.ts

```
@@ -4,18 +4,176 @@ const navigatorDefined = typeof navigator !== 'undefined';
+    }
+
+    // Method 6: Viewport meta tag presence (weight: 1)
+    const hasViewportMeta = document.querySelector('meta[name="viewport"]') !== null;
+    if (hasViewportMeta && hasSmallScreen) {
+        mobileScore += 1;
+    }
+
+    // Method 7: Check for desktop-specific indicators (negative weight)
+    const hasLargeScreen = screenWidth > 1024 && screenHeight > 768;
+    const hasKeyboard = 'keyboard' in navigator;
+    const hasPointer = window.matchMedia && window.matchMedia('(pointer: fine)').matches;
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)